onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -78,9 +78,9 @@ size_t GetAttentionWorkspaceSize(
                           ((sequence_length * kv_sequence_length) * qk_head_size + kv_sequence_length * v_head_size);
 
   if (fused_runner != nullptr) {
-    // Offsets without padding is B + 1. When we add padding, the size need to increase to 2B + 1.
-    size_t sequenceOffsetBytes = sizeof(int) * (batch_size + 1);
-    return qkv_size + reinterpret_cast<MHARunner*>(fused_runner)->getWorkspaceSize() + sequenceOffsetBytes;
+    // There are batch_size + 1 offsets Without padding (or padding removed), and 2 * batch_size + 1 with padding.
+    size_t sequenceOffsetBytes = sizeof(int) * (2 * batch_size + 1);
+    return qkv_size + reinterpret_cast<FusedMHARunnerFP16v2*>(fused_runner)->getWorkspaceSize() + sequenceOffsetBytes;
```

Comment:
FusedMHARunnerFP16v2 supports padding internally?

Reply:
It supports padded input (batch, sequence, hidden_size), which treats each sequence as two sequences (one is non-padding and another padding), unpadded input (token_count, hidden_size).